### ISSUE TITLE
Panic during labels parsing: reflect: NumField of non-struct type string

### ISSUE THREAD

**Issue by nivekuil** [traefik/traefik#7582]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Either a bad command flag on the traefik container or starting a docker swarm service with a malformed label.

### What did you expect to see?

An informative error message

### What did you see instead?
Traefik dashboard/all downstream services were unreachable, panic in logs
```
time="2020-11-24T01:39:30Z" level=info msg="Configuration loaded from flags."
{"level":"error","msg":"Error in Go routine: reflect: NumField of non-struct type string","time":"2020-11-24T01:39:30Z"}
{"level":"error","msg":"Stack: goroutine 41 [running]:\nruntime/debug.Stack(0xc0001dc000, 0x34c674a, 0x17)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9f\ngithub.com/traefik/traefik/v2/pkg/safe.defaultRecoverGoroutine(0x2cc5ae0, 0xc000aa79d0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:66 +0xb4\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1.1(0xc000a89db8)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:74 +0x56\npanic(0x2cc5ae0, 0xc000aa79d0)\n\t/usr/local/go/src/runtime/panic.go:969 +0x1b9\nreflect.(*rtype).NumField(0x2cc5ae0, 0xc000ab61a0)\n\t/usr/local/go/src/reflect/type.go:974 +0xb7\ngithub.com/traefik/paerser/parser.metadata.findTypedField(0x348c675, 0x5, 0x1, 0x3a24720, 0x2cc5ae0, 0xc000ad8c60, 0x0, 0x0, 0x0, 0x0, ...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:134 +0xa8\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc000ad8c60, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:65 +0xe5\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc000ad8bd0, 0x2ec8800, 0x56)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2ee0080, 0xc000ad8b40, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:123 +0x755\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x324ff01, 0x3a24720, 0x2ee0080, 0xc000ad8ab0, 0x2e1d04e, 0x4a)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x3a24701, 0x3a24720, 0x2edf900, 0xc000ad8a20, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:108 +0x4f9\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x317a401, 0x3a24720, 0x2edf900, 0xc000ad8990, 0x2d9a294, 0x41)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc000ad8990, 0x37, 0x4)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:92 +0x2ee\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc000ad8900, 0xc000279680, 0xc000ad8900)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.Add(0x348c675, 0x5, 0x1, 0x2edf580, 0xc000acd680, 0xc000ad8900, 0xc000ad8900, 0x0)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:42 +0xe5\ngithub.com/traefik/paerser/parser.AddMetadata(...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:18\ngithub.com/traefik/paerser/parser.Decode(0xc000aa42a0, 0x2edf580, 0xc000acd680, 0x34926f2, 0x7, 0xc000a89418, 0x3, 0x3, 0x3a232e0, 0xc000a89448)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/parser.go:17 +0xc6\ngithub.com/traefik/traefik/v2/pkg/config/label.DecodeConfiguration(0xc000aa42a0, 0xc000ae0c90, 0x3a232e0, 0xc0001ddc70)\n\t/go/src/github.com/traefik/traefik/pkg/config/label/label.go:17 +0x176\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).buildConfiguration(0xc0004061c0, 0x39ea460, 0xc00067c3f0, 0xc000808400, 0xa, 0x10, 0xa)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/config.go:31 +0x33a\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1.1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:226 +0x68b\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:78 +0x5a\ngithub.com/cenkalti/backoff/v4.RetryNotifyWithTimer(0xc00066a290, 0x7f07d97a0a80, 0xc000452bc0, 0xc000a89f40, 0x39cbe20, 0xc00011c040, 0x0, 0x0)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:52 +0x102\ngithub.com/cenkalti/backoff/v4.RetryNotify(...)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:31\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1(0x39ea3a0, 0xc000924380)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:326 +0x2ed\ngithub.com/traefik/traefik/v2/pkg/safe.(*Pool).GoCtx.func1()\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:36 +0x64\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1(0x35d7768, 0xc000452ba0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:59 +0x4f\ncreated by github.com/traefik/traefik/v2/pkg/safe.GoWithRecover\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:53 +0x49\n","time":"2020-11-24T01:39:30Z"}
{"level":"error","msg":"Provider connection error panic in operation: %!w(\u003cnil\u003e), retrying in 552.330144ms","providerName":"docker","time":"2020-11-24T01:39:30Z"}
{"level":"error","msg":"Error in Go routine: reflect: NumField of non-struct type string","time":"2020-11-24T01:39:31Z"}
{"level":"error","msg":"Stack: goroutine 41 [running]:\nruntime/debug.Stack(0xc0001dc000, 0x34c674a, 0x17)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9f\ngithub.com/traefik/traefik/v2/pkg/safe.defaultRecoverGoroutine(0x2cc5ae0, 0xc00066b8f0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:66 +0xb4\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1.1(0xc000b45db8)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:74 +0x56\npanic(0x2cc5ae0, 0xc00066b8f0)\n\t/usr/local/go/src/runtime/panic.go:969 +0x1b9\nreflect.(*rtype).NumField(0x2cc5ae0, 0xc000ab6020)\n\t/usr/local/go/src/reflect/type.go:974 +0xb7\ngithub.com/traefik/paerser/parser.metadata.findTypedField(0x348c675, 0x5, 0x1, 0x3a24720, 0x2cc5ae0, 0xc00080ebd0, 0x0, 0x0, 0x0, 0x0, ...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:134 +0xa8\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc00080ebd0, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:65 +0xe5\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc00080eb40, 0x2ec8800, 0x56)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2ee0080, 0xc00080eab0, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:123 +0x755\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x324ff01, 0x3a24720, 0x2ee0080, 0xc00080ea20, 0x2e1d04e, 0x4a)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x3a24701, 0x3a24720, 0x2edf900, 0xc00080e990, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:108 +0x4f9\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x317a401, 0x3a24720, 0x2edf900, 0xc00080e900, 0x2d9a294, 0x41)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc00080e900, 0x37, 0x4)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:92 +0x2ee\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc00080e870, 0xc0000c4f80, 0xc00080e870)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.Add(0x348c675, 0x5, 0x1, 0x2edf580, 0xc000acda00, 0xc00080e870, 0xc00080e870, 0x0)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:42 +0xe5\ngithub.com/traefik/paerser/parser.AddMetadata(...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:18\ngithub.com/traefik/paerser/parser.Decode(0xc000bd78c0, 0x2edf580, 0xc000acda00, 0x34926f2, 0x7, 0xc000b45418, 0x3, 0x3, 0x3a232e0, 0xc000b45448)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/parser.go:17 +0xc6\ngithub.com/traefik/traefik/v2/pkg/config/label.DecodeConfiguration(0xc000bd78c0, 0xc0009083c0, 0x3a232e0, 0xc000236460)\n\t/go/src/github.com/traefik/traefik/pkg/config/label/label.go:17 +0x176\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).buildConfiguration(0xc0004061c0, 0x39ea460, 0xc00067c3f0, 0xc000166400, 0xb, 0x10, 0xb)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/config.go:31 +0x33a\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1.1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:226 +0x68b\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:78 +0x5a\ngithub.com/cenkalti/backoff/v4.RetryNotifyWithTimer(0xc00066a290, 0x7f07d97a0a80, 0xc000452bc0, 0xc000b45f40, 0x39cbe20, 0xc00011c040, 0x0, 0x0)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:52 +0x102\ngithub.com/cenkalti/backoff/v4.RetryNotify(...)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:31\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1(0x39ea3a0, 0xc000924380)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:326 +0x2ed\ngithub.com/traefik/traefik/v2/pkg/safe.(*Pool).GoCtx.func1()\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:36 +0x64\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1(0x35d7768, 0xc000452ba0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:59 +0x4f\ncreated by github.com/traefik/traefik/v2/pkg/safe.GoWithRecover\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:53 +0x49\n","time":"2020-11-24T01:39:31Z"}
{"level":"error","msg":"Provider connection error panic in operation: %!w(\u003cnil\u003e), retrying in 1.080381816s","providerName":"docker","time":"2020-11-24T01:39:31Z"}
{"level":"error","msg":"Error in Go routine: reflect: NumField of non-struct type string","time":"2020-11-24T01:39:32Z"}
{"level":"error","msg":"Stack: goroutine 41 [running]:\nruntime/debug.Stack(0xc0001dc000, 0x34c674a, 0x17)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9f\ngithub.com/traefik/traefik/v2/pkg/safe.defaultRecoverGoroutine(0x2cc5ae0, 0xc0006267b0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:66 +0xb4\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1.1(0xc000b45db8)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:74 +0x56\npanic(0x2cc5ae0, 0xc0006267b0)\n\t/usr/local/go/src/runtime/panic.go:969 +0x1b9\nreflect.(*rtype).NumField(0x2cc5ae0, 0xc000e31800)\n\t/usr/local/go/src/reflect/type.go:974 +0xb7\ngithub.com/traefik/paerser/parser.metadata.findTypedField(0x348c675, 0x5, 0x1, 0x3a24720, 0x2cc5ae0, 0xc000727b00, 0x0, 0x0, 0x0, 0x0, ...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:134 +0xa8\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc000727b00, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:65 +0xe5\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc000727a70, 0x2ec8800, 0x56)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2ee0080, 0xc0007279e0, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:123 +0x755\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x324ff01, 0x3a24720, 0x2ee0080, 0xc000727950, 0x2e1d04e, 0x4a)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x3a24701, 0x3a24720, 0x2edf900, 0xc0007278c0, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:108 +0x4f9\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x317a401, 0x3a24720, 0x2edf900, 0xc000727830, 0x2d9a294, 0x41)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc000727830, 0x37, 0x4)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:92 +0x2ee\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc0007277a0, 0xc0005c4a80, 0xc0007277a0)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.Add(0x348c675, 0x5, 0x1, 0x2edf580, 0xc000119f60, 0xc0007277a0, 0xc0007277a0, 0x0)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:42 +0xe5\ngithub.com/traefik/paerser/parser.AddMetadata(...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:18\ngithub.com/traefik/paerser/parser.Decode(0xc0009086c0, 0x2edf580, 0xc000119f60, 0x34926f2, 0x7, 0xc000b45418, 0x3, 0x3, 0x3a232e0, 0xc000b45448)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/parser.go:17 +0xc6\ngithub.com/traefik/traefik/v2/pkg/config/label.DecodeConfiguration(0xc0009086c0, 0xc000aa57d0, 0x3a232e0, 0xc0002288c0)\n\t/go/src/github.com/traefik/traefik/pkg/config/label/label.go:17 +0x176\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).buildConfiguration(0xc0004061c0, 0x39ea460, 0xc00067c3f0, 0xc0005e6000, 0xb, 0x10, 0xb)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/config.go:31 +0x33a\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1.1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:226 +0x68b\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:78 +0x5a\ngithub.com/cenkalti/backoff/v4.RetryNotifyWithTimer(0xc00066a290, 0x7f07d97a0a80, 0xc000452bc0, 0xc000b45f40, 0x39cbe20, 0xc00011c040, 0x0, 0x0)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:52 +0x102\ngithub.com/cenkalti/backoff/v4.RetryNotify(...)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:31\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1(0x39ea3a0, 0xc000924380)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:326 +0x2ed\ngithub.com/traefik/traefik/v2/pkg/safe.(*Pool).GoCtx.func1()\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:36 +0x64\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1(0x35d7768, 0xc000452ba0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:59 +0x4f\ncreated by github.com/traefik/traefik/v2/pkg/safe.GoWithRecover\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:53 +0x49\n","time":"2020-11-24T01:39:32Z"}
{"level":"error","msg":"Provider connection error panic in operation: %!w(\u003cnil\u003e), retrying in 1.31013006s","providerName":"docker","time":"2020-11-24T01:39:32Z"}
{"level":"error","msg":"Error in Go routine: reflect: NumField of non-struct type string","time":"2020-11-24T01:39:34Z"}
{"level":"error","msg":"Stack: goroutine 41 [running]:\nruntime/debug.Stack(0xc0001dc000, 0x34c674a, 0x17)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9f\ngithub.com/traefik/traefik/v2/pkg/safe.defaultRecoverGoroutine(0x2cc5ae0, 0xc00042a390)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:66 +0xb4\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1.1(0xc000b45db8)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:74 +0x56\npanic(0x2cc5ae0, 0xc00042a390)\n\t/usr/local/go/src/runtime/panic.go:969 +0x1b9\nreflect.(*rtype).NumField(0x2cc5ae0, 0xc000168660)\n\t/usr/local/go/src/reflect/type.go:974 +0xb7\ngithub.com/traefik/paerser/parser.metadata.findTypedField(0x348c675, 0x5, 0x1, 0x3a24720, 0x2cc5ae0, 0xc0005f19e0, 0x0, 0x0, 0x0, 0x0, ...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:134 +0xa8\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc0005f19e0, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:65 +0xe5\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2cc5ae0, 0xc0005f1950, 0x2ec8800, 0x56)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x2cc5a01, 0x3a24720, 0x2ee0080, 0xc0005f18c0, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:123 +0x755\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x324ff01, 0x3a24720, 0x2ee0080, 0xc0005f1830, 0x2e1d04e, 0x4a)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x3a24701, 0x3a24720, 0x2edf900, 0xc0005f17a0, 0x0, 0x3a24720)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:108 +0x4f9\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x317a401, 0x3a24720, 0x2edf900, 0xc0005f1710, 0x2d9a294, 0x41)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.add(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc0005f1710, 0x37, 0x4)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:92 +0x2ee\ngithub.com/traefik/paerser/parser.metadata.browseChildren(0x348c675, 0x5, 0x1, 0x3a24720, 0x2edf580, 0xc0005f1680, 0xc0005c5880, 0xc0005f1680)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:47 +0xa5\ngithub.com/traefik/paerser/parser.metadata.Add(0x348c675, 0x5, 0x1, 0x2edf580, 0xc000a8b500, 0xc0005f1680, 0xc0005f1680, 0x0)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:42 +0xe5\ngithub.com/traefik/paerser/parser.AddMetadata(...)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/nodes_metadata.go:18\ngithub.com/traefik/paerser/parser.Decode(0xc00067ced0, 0x2edf580, 0xc000a8b500, 0x34926f2, 0x7, 0xc000b45418, 0x3, 0x3, 0x3a232e0, 0xc000b45448)\n\t/go/pkg/mod/github.com/traefik/paerser@v0.1.0/parser/parser.go:17 +0xc6\ngithub.com/traefik/traefik/v2/pkg/config/label.DecodeConfiguration(0xc00067ced0, 0xc0003b5ce0, 0x3a232e0, 0xc000229c70)\n\t/go/src/github.com/traefik/traefik/pkg/config/label/label.go:17 +0x176\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).buildConfiguration(0xc0004061c0, 0x39ea460, 0xc00067c3f0, 0xc0005e6900, 0xb, 0x10, 0xb)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/config.go:31 +0x33a\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1.1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:226 +0x68b\ngithub.com/traefik/traefik/v2/pkg/safe.OperationWithRecover.func1(0x0, 0x0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:78 +0x5a\ngithub.com/cenkalti/backoff/v4.RetryNotifyWithTimer(0xc00066a290, 0x7f07d97a0a80, 0xc000452bc0, 0xc000b45f40, 0x39cbe20, 0xc00011c040, 0x0, 0x0)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:52 +0x102\ngithub.com/cenkalti/backoff/v4.RetryNotify(...)\n\t/go/pkg/mod/github.com/cenkalti/backoff/v4@v4.0.2/retry.go:31\ngithub.com/traefik/traefik/v2/pkg/provider/docker.(*Provider).Provide.func1(0x39ea3a0, 0xc000924380)\n\t/go/src/github.com/traefik/traefik/pkg/provider/docker/docker.go:326 +0x2ed\ngithub.com/traefik/traefik/v2/pkg/safe.(*Pool).GoCtx.func1()\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:36 +0x64\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1(0x35d7768, 0xc000452ba0)\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:59 +0x4f\ncreated by github.com/traefik/traefik/v2/pkg/safe.GoWithRecover\n\t/go/src/github.com/traefik/traefik/pkg/safe/routine.go:53 +0x49\n","time":"2020-11-24T01:39:34Z"}
{"level":"error","msg":"Provider connection error panic in operation: %!w(\u003cnil\u003e), retrying in 1.582392691s","providerName":"docker","time":"2020-11-24T01:39:34Z"}


```

### Output of `traefik version`: (_What version of Traefik are you using?_)
2.3.3

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```
["--log.format=json" "--providers.docker=true" "--providers.docker.swarmMode=true" "--providers.docker.exposedByDefault=false" "--providers.docker.endpoint=unix:///var/run/docker.sock" "--entrypoints.web.address=:80" "--entrypoints.websecure.address=:443" "--metrics.prometheus=true"]
```

**Comment by ldez** [comment]
Hello,

could you provide your labels, thank you

**Comment by nivekuil** [comment]
Ah, I was going to take a closer look but I see you've already fixed it. I think the square brackets are the problem -- I'm generating these labels from clojure and I think a vector `[1 2 3...]` may have snuck in there.

**Comment by traefiker** [comment]
Closed by #7583.
